File: PaED62__java/PaED62.java
import java.io.IOException;

public class PaED62
{
	public static String hash(String sInput)
	{
		char[] result = new String("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz").toCharArray();

		for (int j = 0; j < sInput.length(); ++j)
			for (int i = 0; i < result.length; ++i)
				result[i] = (char)((((int)result[i] * (i + 1)) + ((int)sInput.charAt(j) * (j + 1))) % (126 - 33) + 33);

		return new String(result);
	}

	public static void main(String[] args)
	{
		System.out.println(hash("Hello, world!"));
		System.out.println(hash("Hello, world! "));

		try
		{
			System.in.read();
		}
		catch (IOException e) {e.printStackTrace();}
	}
}

File: PaED62__cs/PaED62.cs
using System;

class PaED62
{
	static string hash(string sInput)
	{
		char[] sResult = new string("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz").ToCharArray();

		for (int j = 0; j < sInput.Length; ++j)
			for (int i = 0; i < sResult.Length; ++i)
				sResult[i] = (char)((((int)sResult[i] * (i + 1)) + ((int)sInput[j] * (j + 1))) % (126 - 33) + 33);

		return new string(sResult);
	}

	static void Main(string[] args)
	{
		Console.WriteLine(hash("Hello, world!"));
		Console.WriteLine(hash("Hello, world! "));

		Console.ReadKey();
	}
}